packages/instantsearch.js/src/middlewares/createInsightsMiddleware.ts

```diff
@@ -33,32 +42,55 @@ export type InsightsProps<
   onEvent?: (event: InsightsEvent, insightsClient: TInsightsClient) => void;
 };
 
+const ALGOLIA_INSIGHTS_SRC =
+  'https://cdn.jsdelivr.net/npm/search-insights@2.3.0/dist/search-insights.min.js';
+
 export type CreateInsightsMiddleware = typeof createInsightsMiddleware;
 
 export function createInsightsMiddleware<
   TInsightsClient extends null | InsightsClient
->(props: InsightsProps<TInsightsClient>): InternalMiddleware {
+>(props: InsightsProps<TInsightsClient> = {}): InternalMiddleware {
   const {
     insightsClient: _insightsClient,
     insightsInitParams,
     onEvent,
-  } = props || {};
+  } = props;
+
+  let insightsClient: InsightsClient = _insightsClient || noop;
+
   if (_insightsClient !== null && !_insightsClient) {
-    if (__DEV__) {
-      throw new Error(
-        "The `insightsClient` option is required if you want userToken to be automatically set in search calls. If you don't want this behaviour, set it to `null`."
-      );
-    } else {
-      throw new Error(
-        'The `insightsClient` option is required. To disable, set it to `null`.'
-      );
-    }
```

Comment:
effect of this PR, this error no longer happens, even when the auto-load fails, as it could well be something the user has no awareness of.

This means unfortunately you could silently send no events at all.

Reply:
Starting from the version we release this:
- if customers instantiate the middleware with a nooped insights client → we do nothing, it won't send events (but this is expected)
- if customers instantiate the middleware with an insights client → we do nothing, it will send events
- if customers instantiate the middleware without an insights client → we download it for them, it will send events
- if customers don't instantiate the middleware → we instantiate the middleware and download the client, it will send events

Am I missing a scenario?

Reply:
Yes, that's correct. We used to imply somewhat that people should add an insights client or manually pass `null` or a `noop`. As `undefined` (most likely argument) now loads automatically, we don't error in that case anymore

Reply:
One way we could keep this working is if we're able to differentiate the user passing the middleware vs. us passing the middleware. We could have a an internal `createInsightsMiddleware` (not exported) that we use and allows undefined client, and the one users use that still doesn't.

Reply:
but then, would you only auto-load for the default implementation? it means that if someone wants to do a small change (eg filter out a certain widget, or pass initParams), they need to load insights themselves.

Reply:
Isn't it already the case?

Reply:
That's my point. We could preserve the existing behavior of _not_ accepting that the user passed an undefined `insightsClient`. There's no reason for them to do so if they're manually setting up the middleware, right?